### PR TITLE
Setup sbt-typelevel

### PR DESCRIPTION
**Pull request by armanbilge**
As discussed in https://github.com/circe/circe/issues/1958#issuecomment-1134828260.

~~Currently the website build is broken for unclear reasons (besides it's behind a major version in mdoc). I'm loathe to sink time into it if it's going to be replaced by the sbt-typelevel-site plugin.~~ Edit: maybe I fixed it ...

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -1,133 +1,44 @@
-import ReleaseTransformations._
 import microsites.ExtraMdFileConfig
 import microsites.ConfigYml
 import sbtcrossproject.{ CrossProject, CrossType }
-import scala.xml.{ Elem, Node => XmlNode, NodeSeq => XmlNodeSeq }
-import scala.xml.transform.{ RewriteRule, RuleTransformer }
 
-def Scala3 = "3.1.0"
+ThisBuild / tlBaseVersion := "0.14"
+ThisBuild / tlCiReleaseTags := false
```
Comment (armanbilge) [inline build.sbt:6]: This disables CI release, but say the word (and configure the [secrets](https://typelevel.org/sbt-typelevel/secrets.html)) and I'll switch it on :)

**Comment by armanbilge** [comment]
@zarthross we cannot update to sbt-git 2.0.0 here, because of the sbt-microsites plugin. As proposed in https://github.com/circe/circe/issues/1958#issuecomment-1134828260 switching to sbt-typelevel-site would solve that issue.

**Comment by armanbilge** [comment]
@zarthross polite bump on this :) build overhauls tend to have large surface area for conflict.

**Review by zarthross — APPROVED** [review]

**Comment by zarthross** [comment]
I think it looks fine.  I'll have @zmccoy look over it too before we merge it in.  Thanks for taking this on @armanbilge, and sorry for the late review.  

@armanbilge If you still feel like switching this over to sbt-typelevel-site i would welcome the change.

**Comment by armanbilge** [comment]
No problem, thanks for reviewing! Sure, I'll work on the site in a follow-up PR :)

**Review by zmccoy — APPROVED** [review]
This looks great!  Sorry this took a bit for some attention to get to it.  

Thank you for doing this!

**Pull request merged**